### PR TITLE
adding returnurl to logoff

### PR DESCRIPTION
**Pull request by DavidStania**
Adding returnUrl to logoff.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs [hunk 1]
```diff
@@ -243,12 +243,12 @@ public async Task<IActionResult> Login(LoginViewModel model, string returnUrl =
 
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public async Task<IActionResult> LogOff()
+        public async Task<IActionResult> LogOff(string returnUrl)
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs:246]:
```suggestion
        public async Task<IActionResult> LogOff(string returnUrl = null)
```

File: src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs [hunk 2]
```diff
@@ -243,12 +243,12 @@ public async Task<IActionResult> Login(LoginViewModel model, string returnUrl =
...
         {
             await _signInManager.SignOutAsync();
             _logger.LogInformation(4, "User logged out.");
 
-            return Redirect("~/");
+            return RedirectToLocal(returnUrl.ToUriComponents());
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs:251]:
```suggestion
            return RedirectToLocal(returnUrl);
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Comment by sebastienros** [comment]
@hishamco did some changes

**Comment by hishamco** [comment]
The build is failed, why there's two arguments instead of one?

**Comment by jtkech** [comment]
@sebastienros added in the private `RedirectToLocal()` defined here our custom `Redirect` extension that uses `.ToUriComponents()` internally, but this extension is defined in `OC.Mvc.Core` that is not referenced by this project.

Hmm I would have used our `LocalRedirect()` extension instead because in `RedirectToLocal()` we test `IsLocalUrl()`  but for the same reason for now we can't.

So need to use one of the mvc methods by using explicitly `ToUriComponents()`.

    Redirect(url.ToUriComponents());

Or I think as we know it is a local url.

    LocalRedirect(localUrl.ToUriComponents());

**Comment by jtkech** [comment]
Also, not important but I don't think we need to use `returnUrl = null` normally if no returnUrl is passed the mvc binder let it to null.

**Comment by sebastienros** [comment]
Did the change in GH directly, please fix the build and merge.

**Comment by jtkech** [comment]
Okay will do

**Comment by DavidStania** [comment]
Nice! Thank you all.

**Pull request merged**